packages/webpack-extraction-plugin/src/parseCSSRules.ts

```diff
@@ -0,0 +1,50 @@
+import { styleBucketOrdering } from '@griffel/core';
+import { COMMENT, compile, MEDIA, serialize, stringify } from 'stylis';
+
+import type { CSSBucketEntry, CSSRulesByBucket, StyleBucketName } from '@griffel/core';
+
+export function parseCSSRules(css: string) {
+  const cssRulesByBucket = styleBucketOrdering.reduce<CSSRulesByBucket>((acc, styleBucketName) => {
+    acc[styleBucketName] = [];
+
+    return acc;
+  }, {}) as Required<CSSRulesByBucket>;
+  const elements = compile(css);
+  const unrelatedElements: ReturnType<typeof compile> = [];
+
+  let cssBucketName: StyleBucketName | null = null;
+  let cssMeta: Record<string, unknown> | null = null;
+
+  elements.forEach(element => {
+    if (element.type === COMMENT) {
+      if (element.value.indexOf('/** @griffel:css-start') === 0) {
+        cssBucketName = element.value.charAt(24) as StyleBucketName;
+
+        if (cssBucketName === 'm') {
+          cssMeta = JSON.parse(element.value.slice(28, -5));
+        }
+
+        return;
+      }
+
+      if (element.value.indexOf('/** @griffel:css-end') === 0) {
```

Comment:
would `for...of` be a better choice ? then you could simply `break` from the loop. It woudl be clearer than setting the bucket and meta values to `null`

Reply:
`cssMeta` & `cssBucketName` are needed for next iterations in a loop. I am not sure that `for or` will improve it anyhow as there is a need in additional iteration:

```ts
for (let i = 0, l = elements.length; i < l; i++) {
    if (elements[i].type === COMMENT) {
      if (elements[i].value.indexOf('/** @griffel:css-start') === 0) {
        const cssBucketName = elements[i].value.charAt(24) as StyleBucketName;
        const cssMeta: Record<string, unknown> | null =
          cssBucketName === 'm' ? JSON.parse(elements[i].value.slice(28, -5)) : null;

        for (i++; i < l; i++) {
          if (elements[i].value.indexOf('/** @griffel:css-end') === 0) {
            break;
          }

          const cssRule = serialize([elements[i]], stringify);
          const bucketEntry: CSSBucketEntry = cssMeta ? [cssRule, cssMeta] : cssRule;

          cssRulesByBucket[cssBucketName].push(bucketEntry);
        }
      }

      continue;
    }

    unrelatedElements.push(elements[i]);
  }
```

And IMO it's more ugly. Anyway I changed to `for of`.